x/stake/handler_sidechain.go

```diff
@@ -476,6 +480,102 @@ func handleMsgSideChainUndelegate(ctx sdk.Context, msg MsgSideChainUndelegate, k
 	return sdk.Result{Data: finishTime, Tags: tags, Events: events}
 }
 
+func handleMsgSideChainStakeMigration(ctx sdk.Context, msg MsgSideChainStakeMigration, k keeper.Keeper) sdk.Result {
+	if scCtx, err := k.ScKeeper.PrepareCtxForSideChain(ctx, k.DestChainName); err != nil {
+		return ErrInvalidSideChainId(k.Codespace()).Result()
+	} else {
+		ctx = scCtx
+	}
+
+	denom := k.BondDenom(ctx)
+	if msg.Amount.Denom != denom {
+		return ErrBadDenom(k.Codespace()).Result()
+	}
+
+	shares, sdkErr := k.ValidateUnbondAmount(ctx, msg.RefundAddress, msg.Validator, msg.Amount.Amount)
+	if sdkErr != nil {
+		return sdkErr.Result()
+	}
+
+	// unbond immediately
+	ubd, sdkErr := k.BeginUnbonding(ctx, msg.RefundAddress, msg.Validator, shares, false)
+	if sdkErr != nil {
+		return sdkErr.Result()
+	}
+	ubd, events, sdkErr := k.CompleteUnbonding(ctx, ubd.DelegatorAddr, ubd.ValidatorAddr)
+	if sdkErr != nil {
+		return sdkErr.Result()
+	}
+
+	// send coins to pegAccount
+	_, sdkErr = k.BankKeeper.SendCoins(ctx, msg.RefundAddress, sdk.PegAccount, sdk.Coins{ubd.Balance})
```

Comment:
There is a special test case here:
1. user undelegate 1BNB before the hardfork, and it will delay in 7 days.
2. user undeleagate another 1 BNB after the hardfork

what will happen?
a. step #2 will fail?
b. user get 2 BNB undelegated immediately ?